Objects/regions.c

```diff
@@ -1954,3 +1967,39 @@ void _PyRegion_set_cown_parent(PyObject* bridge, PyObject* cown) {
     Py_XINCREF(cown);
     Py_XSETREF(data->cown, cown);
 }
+
+void _Py_RegionRemoveReference(PyObject *src, PyObject *tgt) {
+    if (Py_REGION(src) == Py_REGION(tgt)) {
+        // Nothing to do -- intra-region references have no accounting.
+        return;
+    }
+
+    // If the target is local, then so must the source be. So this should
+    // be covered by the previous check.
+    assert(!_Py_IsLocal(tgt));
+
+    if (_Py_IsImmutable(tgt) || _Py_IsCown(tgt)) {
+        // Nothing to do -- removing a ref to an immutable or a cown has no additional accounting.
+        return;
+    }
+
+    regionmetadata* tgt_md = Py_REGION_DATA(tgt);
+    if (_Py_IsLocal(src)) {
+        // Dec LRC of the previously referenced region
+        // TODO should this decrement be a function, if it hits zero,
+        // then a region could become unreachable.
+        tgt_md->lrc -= 1;
+        return;
+    }
+
+    // This must be a parent reference, so we need to remove the parent reference.
+    regionmetadata* src_md = Py_REGION_DATA(src);
+    regionmetadata* tgt_parent_md = REGION_DATA_CAST(Py_region_ptr(tgt_md->parent));
+    if (tgt_parent_md != src_md) {
+        // TODO: Could `dirty` mean this isn't an error?
+        _PyErr_Region(src, tgt, "(in WB/remove_ref)");
```

Comment:
Dirty should mean that this isn't an error correct. The question is how we deal with these cases. We could still throw an error and require the old parent region to be cleaned beforehand.

Could you make the error message more descriptive?

Reply:
Made the error more descriptive.